include/boost/geometry/formulas/distance_point_segment.hpp

```diff
@@ -33,19 +37,17 @@ template
 <
         typename CT,
         typename Units,
-        template <typename, bool, bool, bool, bool ,bool> class Inverse,
-        template <typename, bool, bool, bool, bool> class Direct,
+        typename FormulaPolicy,
```

Comment:
What is the rationale behind this change? `FormulaPolicy` is conceptually a part of strategies not formulas. I get that this way the template parameter is simpler (even though the typedefs below are more complex) and that this way both inverse and direct formulas are tied together and consistent. But still this change makes the `distance_point_segment` formula's interface different that the interfaces of other formulas. Do you think that we should change the interfaces of all other formulas taking inverse and/or direct formulas now? Even if the answer was yes this change probably shouldn't be a part of this PR, but some different one.

Reply:
Ok, I see now that below you used strategies inside formula. I'm not sure if this is the correct thing to do. In this case I think it'd be better to move this formula to strategies and instead of formula it should be a part of a strategy.